zeroex/orderwatch/watcher.go

```diff
@@ -504,7 +506,12 @@ func (w *Watcher) unwatchOrder(order *meshdb.Order) {
 		}).Error("Failed to update order")
 	}
 
-	w.expirationWatcher.Remove(order.SignedOrder.ExpirationTimeSeconds.Int64(), order.Hash)
+	// If the order is being unwatched because it has expired, it will have been removed
+	// from the ExpirationWatcher immediately so that it can continue checking the remaining
+	// orders immediately
+	if !didExpire {
```

Comment:
"If the order is being unwatched because it has expired" is followed by `if !didExpire`. The comment is describing the opposite condition of the if statement. If this is intentional and not a bug, maybe we can reword the comment? Something like:

```
// If we unwatched the order but it didn't expire, we need to remove it from the expiration watcher. If
// it did expire, it will have already been removed so we don't need to remove it again.
```

Reply:
Not a bug, just badly worded. I like your wording, thanks for that! 🙏